### PR TITLE
Adds the rationale/spec document from Google Docs.

### PR DESCRIPTION
**Pull request by arokem**
As suggested by @Lestropie [here](https://github.com/frheault/tractography_file_format/issues/1#issuecomment-727682011), this moves the rationale/spec document from Google Docs here. I am not sure what to do about all the comments (many quite valuable). I am thinking of ways we could capture all of those as well, and I am open to suggestions.

### CODE REVIEW

**Comment by jdtournier** [comment]
Can I ask whether this is to be a record of what was proposed at the time, or a working document to be updated as consensus emerges? I don't mind the former, but I think it should then be clearly labelled as such, or better yet, be recorded as an issue here. If the latter (which I think it should be), then I think we could merge this as-is to ensure it's recorded in the history, but fairly rapidly trim it down to the essentials. If it's a document to succinctly motivate the current proposal, I don't think it needs to necessarily go into the details of extending other formats, for instance. It should be enough to simply state that none of the existing formats tick enough of the feature boxes to be considered (along with all the difficulties that inevitably arise from trying to maintain backward compatibility, etc). 

Basically, I think the documents on the repo itself should reflect the current position of the proposal, and the motivation page should be positive report of the features we all agree are needed and how the proposal is designed to meet these requirements. I'm just thinking that people coming over to take a quick look will want an overview of the rationale for the _current_ proposal here, not necessarily a precis of all the other options that have been considered - I think these are better discussed and recorded as separate issues.

**Comment by frheault** [comment]
@jdtournier When I created this repo it was about potential proposals and generic discussion. For now TRX is the only one, but it was not my intention to be only about the TRX proposal. It was kind of just an external, independent hub for file format discussion.
In that intention, this file could be here for the record.

I think that this document could stay in the main folder (or maybe create a new one for external general documents). And just like the specification is in the trx_file_memmap/README.md the document specific to a proposal could be in their respective folder.

And at some point when the community feels like no other proposal will emerge, maybe we will modify the repo organization.

**Comment by jdtournier** [comment]
> I think that this document could stay in the main folder (or maybe create a new one for external general documents). And just like the specification is in the trx_file_memmap/README.md the document specific to a proposal could be in their respective folder.

OK, I see what you mean. I agree it's much better to allow different alternative proposals to co-exist before making any actual decisions. That sounds like a good approach, feel free to ignore my previous comment!

**Comment by arokem** [comment]
I think that it would be good to have two documents: one that lays out the desiderata and the process, and another that is a more detailed specification. I was hoping to start with the google doc and then cull it down to fit the former. And as suggested by @frheault, the latter can go into the folders alongside the code.

**Comment by frheault** [comment]
@arokem Is that you mean by specification ?
https://github.com/frheault/tractography_file_format/blob/master/trx_file_memmap/README.md

**Comment by arokem** [comment]
Yes. Exactly! I think that document should describe the format to a degree that someone could read it and implement it in a new language. Something like that.

**Comment by Lestropie** [comment]
Regarding the original comment: I was primarily interested in the specification much more so than the rationale / discussion by which that specification was reached.

The content currently residing at:
https://github.com/frheault/tractography_file_format/blob/master/trx_file_memmap/specifications.md
is actually a mixture of format specification (which is independent of implementation) and sample Python implementation. So splitting that file in two and placing the resulting files appropriately would largely resolve the original concern. E.g. if #10 is adopted and the header is changed from YAML to JSON, that should be reflected in a PR in a change in the specification document, any more verbose documentation / descriptions, and any sample implementations where that change is relevant. But one wouldn't expect to go hunting in the "`trx_file_memmap/`" directory to update the specification document.

As far as the rationale is concerned, a direct copy of that shared document into the repository might indeed cause issues. It also has a lot of content that is not specific to this particular format proposal. It might be better to leave that where it is, and focus instead on a document specifically for this repository that summarises the design decisions / motivations for / strengths and weaknesses of this particular proposal, only contrasting against alternatives where necessary.

**Comment by arokem** [comment]
This will be managed separately in https://github.com/tee-ar-ex/trx-spec, closing here.
